packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -0,0 +1,296 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as kms from '@aws-cdk/aws-kms';
+import {Construct, IResource, Resource} from "@aws-cdk/core";
+import {CfnFileSystem, CfnMountTarget} from "./efs.generated";
+
+// tslint:disable: max-line-length
+/**
+ * EFS Lifecycle Policy, if a file is not accessed for given days, it will move to EFS Infrequent Access.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-elasticfilesystem-filesystem-lifecyclepolicies
+ */
+export enum EfsLifecyclePolicyProperty {
+  /**
+   * After 7 days of not being accessed.
+   */
+  AFTER_7_DAYS,
+
+  /**
+   * After 14 days of not being accessed.
+   */
+  AFTER_14_DAYS,
+
+  /**
+   * After 30 days of not being accessed.
+   */
+  AFTER_30_DAYS,
+
+  /**
+   * After 60 days of not being accessed.
+   */
+  AFTER_60_DAYS,
+
+  /**
+   * After 90 days of not being accessed.
+   */
+  AFTER_90_DAYS
+}
+
+/**
+ * EFS Performance mode.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-efs-filesystem-performancemode
+ */
+export enum EfsPerformanceMode {
+  /**
+   * This is the general purpose performance mode for most file systems.
+   */
+  GENERAL_PURPOSE = "generalPurpose",
+
+  /**
+   * This performance mode can scale to higher levels of aggregate throughput and operations per second with a
+   * tradeoff of slightly higher latencies.
+   */
+  MAX_IO = "maxIO"
+}
+
+/**
+ * EFS Throughput mode.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-elasticfilesystem-filesystem-throughputmode
+ */
+export enum EfsThroughputMode {
+  /**
+   *  This mode on Amazon EFS scales as the size of the file system in the standard storage class grows.
+   */
+  BURSTING = "bursting",
+
+  /**
+   * This mode can instantly provision the throughput of the file system (in MiB/s) independent of the amount of data stored.
+   */
+  PROVISIONED = "provisioned"
+}
+
+/**
+ * Interface to implement AWS File Systems.
+ */
+export interface IEfsFileSystem extends IResource, ec2.IConnectable {
```

Comment:
Wait, `IResource` doesn't need to be in here. Sorry missed that on the first go-around.

Reply:
Just for my understanding, why don't we need this interface to be implemented here? I believe that the file system act as a cloud formation resource.

Reply:
Nevermind, I figured that we are already extending `Resource` which implements `IResource`.